worlds/yoshisisland/__init__.py

```diff
@@ -0,0 +1,400 @@
+        self.multiworld.get_location("Ride Like The Wind: Gather Coins", self.player).place_locked_item(self.create_item("Bandit Consumables"))
+
+    def place_locked_item(self, excluded_items: Set[str], location: str, item: str) -> None:
+        excluded_items.add(item)
+
+        item = self.create_item(item)
+
+        self.multiworld.get_location(location, self.player).place_locked_item(item)
+
+    def generate_early(self):
+        setup_gamevars(self, self.multiworld, self.player)
+
+    
+    def get_excluded_items(self, multiworld: MultiWorld, player: int) -> Set[str]:
+        excluded_items: Set[str] = set()
+
+        starting_gate = ["World 1 Gate", "World 2 Gate", "World 3 Gate", "World 4 Gate", "World 5 Gate", "World 6 Gate"]
+
+        excluded_items.add(starting_gate[multiworld.starting_world[player].value])
+
+        if self.multiworld.shuffle_midrings[self.player] == 0:
+            excluded_items.add('Middle Ring')
+
+        if self.multiworld.add_secretlens[self.player] == 0:
+            excluded_items.add('Secret Lens')
+
+        if self.multiworld.extras_enabled[self.player] == 0:
+            excluded_items.add('Extra Panels')
+            excluded_items.add('Extra 1')
+            excluded_items.add('Extra 2')
+            excluded_items.add('Extra 3')
+            excluded_items.add('Extra 4')
+            excluded_items.add('Extra 5')
+            excluded_items.add('Extra 6')
+
+        if self.multiworld.split_extras[self.player] == 1:
+            excluded_items.add('Extra Panels')
+        else:
+            excluded_items.add('Extra 1')
+            excluded_items.add('Extra 2')
+            excluded_items.add('Extra 3')
+            excluded_items.add('Extra 4')
+            excluded_items.add('Extra 5')
+            excluded_items.add('Extra 6')
+
+        if self.multiworld.split_bonus[self.player] == 1:
+            excluded_items.add('Bonus Panels')
+        else:
+            excluded_items.add('Bonus 1')
+            excluded_items.add('Bonus 2')
+            excluded_items.add('Bonus 3')
+            excluded_items.add('Bonus 4')
+            excluded_items.add('Bonus 5')
+            excluded_items.add('Bonus 6')
+
+        return excluded_items
+
+    def create_item_with_correct_settings(self, multiworld: MultiWorld, player: int, name: str) -> Item:
+        data = item_table[name]
+        if data.useful:
+            classification = ItemClassification.useful
+        elif data.progression:
+            classification = ItemClassification.progression
+        elif data.trap:
+            classification = ItemClassification.trap
+        else:
+            classification = ItemClassification.filler
+        item = Item(name, classification, data.code, player)
+
+        if not item.advancement:
+            return item
+
+        if (name == 'Car Morph' and get_option_value(multiworld, player, "stage_logic") != 0):
```

Comment:
Don't use parantheses around an if condition. Otherwise, every Python programmer will hate you lol